x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/Classification.java

```diff
@@ -406,8 +405,38 @@ public boolean supportsCategoricalFields() {
         return additionalProperties;
     }
 
-    private static Object extractMapping(String path, Map<String, Object> mappingsProperties) {
-        return extractValue(String.join(".properties.", path.split("\\.")), mappingsProperties);
+    // Visible for testing
+    static Object extractMapping(String path, Map<String, Object> mappingsProperties) {
```

Comment:
I'm open to suggestions how this could be simplified.
We just want to fetch mapping by path. Should I be using `FieldCapabilities` API instead?

Reply:
We probably should use field capabilities. 

It returns nice objects that are like:
```
  "data.foo.bar" : {
      "double" : {
        "type" : "double",
        "searchable" : true,
        "aggregatable" : true,
        "indices" : [
          "fake-foo-2"
        ]
      },
      "long" : {
        "type" : "long",
        "searchable" : true,
        "aggregatable" : true,
        "indices" : [
          "fake-foo"
        ]
      }
    },
```
Where `data.foo.bar` would be this `path` variable.

Here is another example:

```
PUT complicated
{
  "mappings": {
    "properties": {
      "foo": {
        "type": "text",
        "fields": {
          "sub_foo": {
          "type": "keyword"
          }
        }
      },
      "bar": {
        "properties": {
          "baz": {
            "type": "long"
          }
        }
      }
    }
  }
}
```

Returns the following in the field caps:
```
{
  "indices" : [
    "complicated"
  ],
  "fields" : {
    "foo.sub_foo" : {
      "keyword" : {
        "type" : "keyword",
        "searchable" : true,
        "aggregatable" : true
      }
    },
    "bar.baz" : {
      "long" : {
        "type" : "long",
        "searchable" : true,
        "aggregatable" : true
      }
    },
    "foo" : {
      "text" : {
        "type" : "text",
        "searchable" : true,
        "aggregatable" : false
      }
    }
    "bar" : {
      "object" : {
        "type" : "object",
        "searchable" : false,
        "aggregatable" : false
      }
    }
  }
}
```

Reply:
Done.